src/main/java/controller/BlackJackController.java

```diff
@@ -0,0 +1,98 @@
+package controller;
+
+import domain.Answer;
+import domain.CardDeck;
+import domain.Game;
+import domain.participant.Dealer;
+import domain.participant.Player;
+import domain.participant.Players;
+import dto.DealerHandsDto;
+import dto.ParticipantDto;
+import dto.ParticipantsDto;
+import view.InputView;
+import view.OutputView;
+
+public class BlackJackController {
+    private final InputView inputView;
+    private final OutputView outputView;
+
+    public BlackJackController(final InputView inputView, final OutputView outputView) {
+        this.inputView = inputView;
+        this.outputView = outputView;
+    }
+
+    public void run() {
+        final Players players = Players.from(inputView.readNames());
+        final CardDeck cardDeck = CardDeck.generate();
+        final Dealer dealer = new Dealer(cardDeck);
+        final Game game = new Game(dealer, players);
+
+        initHands(players, dealer);
+        dealWithPlayers(players, dealer);
+
+        if (!players.isAllBust()) {
+            dealer.deal();
+            printDealerTurnMessage(dealer.countAddedHands());
+        }
+        printFinalResult(players, dealer, game);
+    }
+
+    private void printDealerTurnMessage(final int turn) {
+        for (int i = 0; i < turn; i++) {
+            outputView.printDealerTurnMessage();
+        }
+    }
+
+    private void dealWithPlayers(final Players players, final Dealer dealer) {
+        for (Player player : players.getPlayers()) {
+            deal(player, dealer);
+        }
+    }
```

Comment:
여기서 deal 에게 플레이어와 딜러를 주지 않고도  카드를 주고 받을 수 있게 해볼까요?

```java

class Dealer {
...
    public Card pickCard() {
        return cardDeck. pop();
    }
}
```
```suggestion
    private void dealWithPlayers(final Players players, final Dealer dealer) {
        for (Player player : players.getPlayers()) {
        Card playerCard = dealer.pickCard();
        players.add( playerCard);
        }
    }
```